server/src/main/java/io/crate/execution/dml/upsert/TransportShardUpsertAction.java

```diff
@@ -246,7 +254,9 @@ protected WritePrimaryResult<ShardUpsertRequest, ShardResponse> processRequestIt
 
                 if (!request.continueOnError()) {
                     shardResponse.failure(e);
-                    break;
+                    shortCircuited = true;
+                    // Continue, mark remaining items to skip on replica.
+                    continue;
```

Comment:
Not marking it here because on catch block we already mark as SKIP_ON_REPLICA.

This one is not really needed on master (only insert from values with a single value has continueOnError flipped) but will be needed when we have fail_fast feature